common/pkg/k8s/client/apply.go

```diff
@@ -22,7 +22,6 @@ func ServerSideApply(
 			ctx,
 			objs[i],
 			ctrlclient.Apply,
-			ctrlclient.ForceOwnership,
```

Comment:
AFAIK `controller-runtime` doesn't actually fully support SSA - see discussion in https://github.com/kubernetes-sigs/controller-runtime/issues/347#issuecomment-1470092104.

There is work ongoing to support `ApplyConfigurations` but right now if you do not send all fields and force ownership then there are issues determining whether fields are unset orset-and-zero'd. I'm not completely clear on the implications of this, but I would prefer to keep the existing `ForceOwnership` behaviour for current usage and add the ability to not force ownership for your namespace use case. WDYT?

Reply:
I pushed https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/559 to prove to myself that this does indeed work and it does... which is good, but I still have concerns about this for the general SSA case.

Reply:
Sounds good to me! Let's make it opt-in to `ForceOwnership` by requiring to pass in that option too. It will require changes to existing calls to this function but that's OK.